STYLEGUIDE.md

```diff
@@ -0,0 +1,61 @@
+# Style Guide
+
+This site is built using [docusaurus](https://docusaurus.io/). Docusaurus will produce static HTML pages and a site built
+around [React](https://react.dev/). This page will try to give an overview of Docusaurus' pluses and minuses.
+
+## Not Pure Markdown
+This means Docusaurus can support rendering normal markdown (*.md) files into static
+content, but it is also capable of rendering MDX/JSX into html as well. That is a powerful feature and opens the door
+to some snazzy effects, reuse of content and code, etc. This does come at a small cost of "not being pure markdown".
+
+Why is it considered a 'cost' that it's not pure markdown? That's because the markdown you look at is _supposed_ to be
+legible without processing. Sometimes though, you don't want to reinvent the wheel or copy/paste content etc. That's 
+where MDX/JSX will be amazing but that's also when you'll not be able to "just open the markdown file". So while it's
+often neat and useful, sometimes it's a drag. It also leaves all sorts of "extra" stuff at the top of the file and using
+the React components is somewhat "clunky".
+
+## Expectations for Style and Content
+
+Here is a list of expectations you should follow when writing a page backed by Docusaurus. These are in no particular
+order at this time but should be at least roughly grouped :
+
+* markdown: __DO__ wrap lines or format lines ___you are adding___ Consider wrapping at around 120 chars
+* markdown: __DO NOT__ wrap lines for content ___you are not touching___
+* markdown: __DO NOT__ format or wrap lines for an entire file automatically
+* markdown: __DO__ add [front matter](https://docusaurus.io/docs/next/markdown-features#front-matter) to every markdown
+            page. __Most importantly__, `title` and `description` but optionally consider adding and `image`
+* format: __DO__ format an entire markdown and NOTHING else (if trying to format a document)
+* format: __DO NOT__ format an entire markdown __and change__ content, format content __ONLY__ and with agreement from others
+* links: __DO__ use relative path links to the actual markdown file (../../../docs/some/05-markdown.md)
+* links: __DO NOT__ use fully qualified links pointing to the docs (https://openziti.io/...)
+* links: __DO NOT__ use the Docusaurus 'id' in a link as in: `../../../docs/some/markdown`. (Notice the lack of ".md"/".mdx")
+* images: __DO__ put images into the `/docusaurus/static/img` directory
+* images: __DO__ try to organize and name images properly
+* images: __DO__ try to use `.SVG` files as much as possible
+* images: __DO__ take care to test your image on both light and dark mode to see if it looks acceptable in both views
+* images: __DO__ try to not push too many binary (image) files over and over as it bloats the git repo
+* sidebar: __DO__ try to favor organizing pages using numerical prefixing: `01-page1.md`, `10-page2.md`

```

Comment:
This is supported by Docusaurus, but inferior to sidebar position metadata in practice because it creates work when the index prefix must change thereby breaking file path references in Markdown links, and because the index prefix is subordinate to position metadata, leading to a non-obvious precedence of prefixes and metadata competing for the end result.

Reply:
yep. it's also more clear and easier to see how it sorts. let's hope we don't find ourselves reorganizing things too much.

Reply:
My primary objection is that it breaks links when prefixes change. If we don't need to reorganize things too much that's less of an issue. My secondary objection is that it is less clear, and harder to see how it sorts because the prefixes are subordinate to position metadata. The only way to make prefixes clear is to prohibit the use of position metadata, which is not suggested by these guidelines.

Reply:
it's superior to see how it sorts, it's sorted in the directory. that's the major benefit. the major drawback is that it's a major pain to rearrange things dramatically. that's why, just like the BASIC days, if you leave sensible gaps, rearranging things is not an issue.

I would definitely prefer to prohibit position data, but instead of being overly hamfisted, I figure sensibilities are allowable. using front matter, while flexible, is incredibly difficult to debug. The other option is to explicitly manage the sidebar. If you would prefer to mandate the prohibition of frontmatter from positioning, that's fine with me. i can add that too. i just didn't feel the need for it

Reply:
I have wondered if the autogenerated sidebar is more or less trouble than a manual sidebar. I'm comfortable with leaving both indexing and position metadata as tools in the toolbox. I recommend that we do not recommend indexing. That's overly prescriptive and will create unnecessary churn. Indexing has it's place but I'm not in favor of prescribing that moving forward nor retrofitting  the site with indexed content filenames, heaven forbid.

Reply:
> I'm comfortable with leaving both indexing and position metadata as tools in the toolbox.

I am not. It leads to confusion. Until it creates unnecessary churn, I am going to add it to the guide. People aren't mandated to follow it anyway so we'll see what happens. We will not be refactoring anything that's already out there.